actions/gha-workflow-validator/project.json

```diff
@@ -0,0 +1,46 @@
+{
+  "name": "gha-workflow-validator",
+  "$schema": "../../node_modules/nx/schemas/project-schema.json",
+  "projectType": "application",
+  "targets": {
+    "test": {
+      "executor": "@nx/jest:jest",
+      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
+      "options": {
+        "jestConfig": "actions/gha-workflow-validator/jest.config.ts"
+      }
+    },
+    "build": {
+      "dependsOn": ["build:internal"],
+      "command": "rimraf dist/package.json"
+    },
+    "build:internal": {
+      "executor": "@nx/esbuild:esbuild",
+      "outputs": ["{options.outputPath}"],
+      "defaultConfiguration": "production",
+      "options": {
+        "format": ["cjs"],
+        "thirdParty": true,
+        "outputPath": "actions/gha-workflow-validator/dist/",
+        "main": "actions/gha-workflow-validator/src/index.ts",
```

Comment:
I've tried messing around with the `nx` variables and haven't had much success. I will give it another shot because it is rather annoying having to run the test/build commands only in the root dir of the repo.

Reply:
yeah, that's odd. you should be able to run nx commands within the pkg dir too. 
I tested in the GATI repo, I was able to run `pnpm nx build` in a package dir and it worked fine

Reply:
I'm going to create a follow-up task for this, as well as the prettier setup.